viking/src/tools/check.rs

```diff
@@ -144,34 +154,44 @@ fn check_all(
                 decomp_symtab,
                 function,
             )?;
-            if !ok {
+            if matches!(ok, CheckResult::MismatchError) {
                 failed.store(true, std::sync::atomic::Ordering::Relaxed);
             }
+            else if matches!(ok, CheckResult::MatchWarn) {
+                matching_functions.lock().unwrap().push(function.addr);
+            }
 
             Ok(())
         })
     })?;
 
+    let functions_to_update = matching_functions.lock().unwrap();
+    if update_matching {
+        let mut new_functions = functions.iter().cloned().collect_vec();
+        new_functions
+            .iter_mut()
+            .filter(|info| functions_to_update.contains(&info.addr))
+            .for_each(|info| info.status = functions::Status::Matching);
+
+        functions::write_functions(&new_functions, version)?;
+    }
+
     if failed.load(std::sync::atomic::Ordering::Relaxed) {
         bail!("found at least one error");
     } else {
         Ok(())
     }
 }
 
-fn get_function_to_check_from_args(args: &[String]) -> Result<String> {
-    let mut maybe_fn_to_check: Vec<String> = args
-        .iter()
-        .filter(|s| !s.starts_with('-'))
-        .cloned()
-        .collect();
-
-    ensure!(
-        maybe_fn_to_check.len() == 1,
-        "expected only one function name (one argument that isn't prefixed with '-')"
-    );
+fn get_function_to_check_from_args(args: &[String]) -> Result<Option<&str>> {
```

Comment:
At this point you should probably just use a proper command line parser library instead of duplicating this for every single option you want to add (which is a code smell).

Reply:
Any preference, or should I just pick the first result when googling for it - `clap`? (https://github.com/clap-rs/clap)

Reply:
The requirements for viking are:

* being able to specify flags/options without too much boilerplate
* letting us grab unknown arguments so we can pass them as is to asm-differ (the library shouldn't complain about unknown arguments)
* letting the user pass arguments in any order, e.g. `tools/check -c function_name` should work, but so should `tools/check function_name -c`